sim-lib/src/lnd.rs

```diff
@@ -54,13 +60,61 @@ impl LightningNode for LndNode {
 
     async fn send_payment(
         &self,
-        _dest: PublicKey,
-        _amount_msat: u64,
+        dest: PublicKey,
+        amount_msat: u64,
     ) -> Result<PaymentHash, LightningError> {
-        unimplemented!()
+        let mut client = self.client.clone();
+        let router_client = client.router();
+
+        let amt_msat: i64 = amount_msat
+            .try_into()
+            .map_err(|_| LightningError::SendPaymentError("Invalid send amount".to_string()))?;
+
+        let preimage = PaymentPreimage(rand::random());
+
+        let mut dest_custom_records = HashMap::new();
+        dest_custom_records.insert(KEYSEND_KEY, preimage.0.to_vec());
+
+        let response = router_client
+            .send_payment_v2(SendPaymentRequest {
```

Comment:
Needs timeout (5 mins?) + hash of `preimage`

Reply:
fixed